src/file-picker/src/FilePicker.js

```diff
@@ -108,21 +116,19 @@ export default class FilePicker extends PureComponent {
     let buttonText
     if (files.length === 0) {
       buttonText =
-        buttonLabels === undefined || buttonLabels.selectLabel === undefined
+        selectFileButtonLabel === undefined
           ? 'Select file'
```

Comment:
If you use defaultProps you shouldn't need to do these `undefined` checks which will clean up the code quite a bit.